src/Input/InputEvents.cpp

```diff
@@ -464,8 +470,14 @@ InputEvents::ProcessMenuTimer()
     /* no menu updates while a dialog is visible */
     return;
 
-  if (MenuTimeOut == CommonInterface::GetUISettings().menu_timeout)
+  if (MenuTimeOut == CommonInterface::GetUISettings().menu_timeout) {
     HideMenu();
+    if (LockScreen == 1)   {     ///////// LOCKSCREEN
```

Comment:
More stupid comments instead of explaining what's doing on. I tried, but failed to understand this part of your code.